### PR TITLE
Editorial: Replace mapping table and summary/details with headings

### PR DESCRIPTION
**Pull request by spectranaut**
Githack link, as PR Preview doesn't include styles: https://raw.githack.com/w3c/core-aam/remove-tables/index.html#mapping_role

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/core-aam/pull/180.html" title="Last updated on Aug 21, 2023, 6:32 PM UTC (cdec3fe)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/core-aam/180/5d6012b...cdec3fe.html" title="Last updated on Aug 21, 2023, 6:32 PM UTC (cdec3fe)">Diff</a>

### CODE REVIEW

File: index.html
Comment (spectranaut) [inline index.html:205]: I just moved this code to here from the mapping tables file, I wonder if @pkra has an idea about where is should belong? here and duplicated across specs... or in the aria commons file still?

Also I'm fuzzy on the publication process, I have some vague idea we are suppose to not have javascript in the published draft.

Comment (pkra) [reply index.html:205]: >  I wonder if @pkra has an idea about where is should belong? here and duplicated across specs... or in the aria commons file still?

Given how small it is now that, I personally don't mind either way - whichever the aam editors prefer.

> Also I'm fuzzy on the publication process, I have some vague idea we are suppose to not have javascript in the published draft.

I don't know but would observe that you're using an existing script tag; even if that script tag had "just" the configuration for the tables script, it seems to indicate that JS is ok.

**Comment by pkra** [comment]
My browser is hanging when opening up the larger commit so I can't leave inline comments. 

Here are some things I noticed so far

* the visual styles are mostly gone (summary/details, the table inside). Should we try to recover them?
* the content in the summary elements is repeated in a table caption. 
  * Is this intentional? (e.g., to avoid links in summary) 
  * Also, local references in those table captions appear to be broken

@spectranaut @jnurthen I'm not sure how to make sure that nothing was accidentally dropped. Do you have any thoughts on that?

**Comment by spectranaut** [comment]
> Here are some things I noticed so far
> 
>     * the visual styles are mostly gone (summary/details, the table inside). Should we try to recover them?

Do you mean in the PR Preview? The styles have always been broken here, unfortunately. You can download and look at it locally or use githack: https://raw.githack.com/w3c/core-aam/remove-tables/index.html#mapping_role

I wonder if there is a way to fix this.

>     * the content in the summary elements is repeated in a table caption.
>       * Is this intentional? (e.g., to avoid links in summary)

This is how it was before, but we can change it. The "repeat" was a link to ARIA, but now it seems broken, as you point out below.

>       * Also, local references in those table captions appear to be broken

I'll fix this, it's weird though.. those used to be links back to the ARIA spec. But even in the editors draft they link just back to core-aam, I'm not sure how or when this changed.

> 
> @spectranaut @jnurthen I'm not sure how to make sure that nothing was accidentally dropped. Do you have any thoughts on that?

I wrote a little script to do the translation locally, I'm pretty confident I didn't drop anything! I'd guess we don't have to worry about that.

**Comment by pkra** [comment]
Thanks for clarifying, @spectranaut! (Especially me forgetting about the CSS issue).  

This looks good to me then.

**Comment by pkra** [comment]
FYI the PR preview issue seems to be https://github.com/tobie/pr-preview/issues/87

**Comment by pkra** [comment]
As per ARIA Editors meeting 2023-06-26, @scottaohara wants to coordinate with @spectranaut on this.

**Comment by scottaohara** [comment]
spent time checking this out and while i had mentioned during the editor's call that maybe we could revise the presentation/markup for this component, I'm going to retract those ideas for now.  No reason to hold this up further, and i'm doubtful the benefit of changing this up further at this point.

i'm assuming the broken links to the ARIA spec will be fixed once this gets merged?  The links are setup exactly the same between the published version and this, so I can only assume it's just some problem with respec not building properly?

**Comment by pkra** [comment]
As per ARIA Editors meeting 2023-07-10, @spectranaut will simplify further, dropping summary/details, introducing headings etc.

**Comment by spectranaut** [comment]
The HTML is a little wacky, I'll fix before landing, but @scottaohara and @pkra and @jnurthen what do you think of this radical change we discussed: https://raw.githack.com/w3c/core-aam/remove-tables/index.html#mapping_role

**Comment by scottaohara** [comment]
i think it's an improvement.

tbh i'm going back and forth on whether the individual roles/attributes should take part in the TOC... but it would definitely help with the easier permalinking

**Comment by pkra** [comment]
I'll second @scottaohara -- looks much better to me, too!

Could we drop the table captions as well if we linked the terms in the headings? For roles, we could alternatively just link the computed role entry but we don't have that for states and properties (though I suppose that might happen if we had synonymous properties some day).

**Comment by spectranaut** [comment]
> Could we drop the table captions as well if we linked the terms in the headings? For roles, we could alternatively just link the computed role entry but we don't have that for states and properties (though I suppose that might happen if we had synonymous properties some day).

I was also trying to think of a way to drop the headings. Maybe I'll just add an extra row to all the tables and say "WAI-ARIA reference" in the first column and the relevant linked role or attribute in the second?

**Comment by jcsteh** [comment]
- As an implementer, being able to easily link to the individual mappings is going to be awesome. Thank you.
- This also works just fine for me as a screen reader user.
- Removing summary/details does mean this takes noticeably longer to load with a screen reader. I'm not sure about the load time without. That's probably not a big concern, but I thought it worth noting. It also probably means I have more a11y engine optimisation to do. :)

**Comment by jcsteh** [comment]
> Maybe I'll just add an extra row to all the tables and say "WAI-ARIA reference" in the first column and the relevant linked role or attribute in the second?

That's arguably clearer anyway. Right now, unless you know how the spec is structured, you might not necessarily know what that link is for, since it effectively duplicates the heading label.

File: index.html
Comment (spectranaut) [reply index.html]: removed summary details, so don't need script after all

**Review by scottaohara — APPROVED** [review]
@spectranaut i think this is good to go.

optional change would be to update the spec to _not_ include the individual role/attribute names in the TOC.  but that doesn't need to be part of this change, if it's done at all

**Pull request merged**
